tests/grep/test_grep.py

```diff
@@ -0,0 +1,177 @@
+# -*- coding: utf-8 -*-
+"""tests for the 'grep' command"""
+import os
+
+from stash.tests.stashtest import StashTestCase
+
+
+class GrepTests(StashTestCase):
+    """ Tests for the 'grep' command
+        No test cases for --color and --encoding.
+        Everything else has a test case here. """
+
+    def setUp(self):
+        """setup the tests"""
+        self.cwd = self.get_data_path()
+        StashTestCase.setUp(self)
+
+    def get_data_path(self):
+        """return the data/ sibling path"""
+        return os.path.abspath(os.path.join(os.path.dirname(__file__), "data"))
+
+    def test_grep_case(self):
+        """test 'grep' case sensitivity."""
+        output = self.run_command("grep 'grep' case_content.txt", exitcode=0)
+        self.assertEqual("""test grep lower case test
+test grep Grep GREP grep GREP Grep test
+""",output)
+        output = self.run_command("grep -i 'grep' case_content.txt", exitcode=0)
+        self.assertEqual("""test grep lower case test
+test Grep mixed case test
+test GREP upper case test
+test grep Grep GREP grep GREP Grep test
+""",output)
+
+    def test_grep_count(self):
+        """ test 'grep' counting and file name display"""
+        with self.subTest(flavor='--count single_file'):
```

Comment:
Unfortunately, `subTest` is not available in py2, though I suppose at some point we have to move away from py2 anyway...

Reply:
I converted the the subTest parts all to their own test methods. Therefore this problem is gone as well.